### PR TITLE
Add Named pipe support to solve the problem with elevations

### PR DESCRIPTION
**Pull request by kdschlosser**
If the first running instance of EG is elevated and the second is not windows com throws an error. this will restart EG as elevated if this error occurs.

### CODE REVIEW

**Comment by kdschlosser** [comment]
I am not sure as to why appveyor fails. but when I build it builds fine.

**Comment by kdschlosser** [comment]
please make sure you read the commit comments on the last commit. as i feel this addition can bring some nifty things to EG 

here is some sample code if you want to tell EG to do different things from another program. the code example you run from a python interpreter outside of EG. and this will cause EG to minimize.

    import win32pipe
    import win32file

    def send_message(msg):
        p = win32pipe.CreateNamedPipe(
            r'\\.\pipe\eventghost_pipe',
            win32pipe.PIPE_ACCESS_DUPLEX,
            win32pipe.PIPE_TYPE_MESSAGE | win32pipe.PIPE_WAIT,
            1,
            65536,
            65536,
            300,
            None
        )

        win32pipe.ConnectNamedPipe(p, None)
        win32file.WriteFile(p, msg)

    send_message('eg.document.frame.Iconize, (True,)')

**Comment by topic2k** [comment]
still no luck.
```python
Traceback (most recent call last):
  File "EventGhost.pyw", line 53, in <module>
  File "c:\Program Files (x86)\EventGhost\eg\__init__.py", line 31, in <module>
    import Cli
  File "c:\Program Files (x86)\EventGhost\eg\Cli.py", line 146, in <module>
    args.pluginFile
  File "c:\Program Files (x86)\EventGhost\eg\NamedPipe.py", line 164, in send_message
    None,
pywintypes.error: (5, 'CreateFile', 'Access denied')
```
(EG running as Administrator and normal user doubleclicked on a .egplugin file)

**Comment by kdschlosser** [comment]
I also want you to run EG s a normal user and install a plugin as Administrator to see if this will handle that scenario and well. I am thinking it will.

**Comment by kdschlosser** [comment]
I also wanted to add this as well because i forgot to

Resolves #229

**Comment by kdschlosser** [comment]
One other not on this.... !!! YAY!!!! this is going to help out a HUGE amount if we manage to get this going. 

But check this out. You know how if you go to install EG and you have an instance already and it asks you to close it?. well this is the mechanism to do it automatically.

But i mention this because of #245 and what exactly is getting stuck in the builder i am not sure. I am thinking it is when eg gets run with -install to create the .pyc files. I don't specifically know what the reasoning is behind launching EG to compile those files. because python does this when the file (module) is imported, it doesn't speed anything up.

@topic2k do you know what the purpose to it is?

**Comment by topic2k** [comment]
> @topic2k do you know what the purpose to it is?

no, not really. I can only guess that it was to speed up (first) loading of EG. Maybe in older python version it was a speed improvement.
Secondly it creates a COM server. Could it be needed to have EG be registered to open .egplugin/.egtree?

What if you comment that line in the setup template? (line 208, put a ; in front) Add as commit here and i will test.

**Comment by kdschlosser** [comment]
Fantastic! No more hang when installing EG and the user is in the Administrators group.. woo hoo

**Comment by kdschlosser** [comment]
Everything loads up just fine. there is a very slight performance hit the first time it loads up. but it's really really tiny.

**Comment by kdschlosser** [comment]
all of the COM server methods/functions can get depreciated if we get this named pipe thing going properly. they will become obsolete.

**Comment by kdschlosser** [comment]
I got it working!

File: eg/Classes/App.py [hunk 1]
```diff
@@ -133,6 +133,7 @@ def OnExit(self):
             eg.PrintDebugNotice("The following threads did not terminate:")
             for thread in threads:
                 eg.PrintDebugNotice(" ", thread, thread.getName())
+
```
Comment (topic2k) [inline eg/Classes/App.py]: I think this has nothing to do with the PR and should be left out or made in another PR for code cleanup. Same goes for other whitespace/blank line fixes

**Comment by topic2k** [comment]
fixes #245, fixes #229

**Comment by topic2k** [comment]
Let's say i'm running EG with full elevation. Now from a command line with limited elevation i call `eventghost.exe -restart`. EG will be restarted, but now with limited elevation. Can we check what elevation the running EG has and automatically decide if we restart with full or limited elevation?

**Comment by kdschlosser** [comment]
run eventghost -restart with elevated privlages

File: eg/Classes/App.py (hunk 1 above)
Comment (kdschlosser) [inline eg/Classes/App.py]: HA sorry about that. that happened because i didn't have the thread that runs the named pipe set up as a daemon thread at first and that is where i told the thread to stop. i forgot to remove the extra line. my bad
